### PR TITLE
PLT-7541 Pull plutus scripts into separate repo

### PR DESCRIPTION
**Pull request by jhbertra**
- [x] Remove `marlowe-plutus` project
- [x] Add pre-compiled scripts to data files in `marlowe-cardano` project
- [x] Replace references to marlowe and payout validators with pre-compiled scripts
- [x] Add script to refresh validators

Pre-submit checklist:
- Branch
    - [x] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [x] Formatting, PNG optimization, etc. are updated
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [x] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

**Comment by jhbertra** [comment]
Question - what to do with the test report? The test code is now mostly in the other repo.

File: marlowe-cli/src/Language/Marlowe/CLI/Export.hs [hunk 1]
```diff
@@ -205,10 +212,26 @@ exportMarlowe marloweParams protocolVersion costModel network stake contractFile
...
         hPutStrLn stderr $ "Total size: " <> show (viSize + diSize + riSize)
 
+readMarloweValidator :: (MonadIO m, IsPlutusScriptLanguage lang) => m (PlutusScript lang)
+readMarloweValidator = readValidator "marlowe-semantics.plutus"
+
+readRolePayoutValidator :: (MonadIO m, IsPlutusScriptLanguage lang) => m (PlutusScript lang)
+readRolePayoutValidator = readValidator "marlowe-rolepayout.plutus"
+
+readOpenRoleValidator :: (MonadIO m, IsPlutusScriptLanguage lang) => m (PlutusScript lang)
+readOpenRoleValidator = readValidator "marlowe-open-role.plutus"
```
Comment (jhbertra) [inline marlowe-cli/src/Language/Marlowe/CLI/Export.hs:222]:
```suggestion
readOpenRoleValidator = readValidator "open-role.plutus"
```

**Review by paluh — COMMENTED** [review]
Nice! I really like that you provided diff and refresh for scripts Jamie. Thanks a lot for additional cleanups (wordings etc.).

File: marlowe-test/test/Spec/Marlowe/Marlowe.hs [hunk 2]
```diff
@@ -151,19 +145,6 @@ tests =
...
 
-maxMarloweValidatorSize :: Int
-#ifdef TRACE_PLUTUS
-maxMarloweValidatorSize = 12737
-#else
-maxMarloweValidatorSize = 12194
-#endif
-
--- | Test that the untyped validator is not too large.
-marloweValidatorSize :: IO ()
```
Comment (bwbush) [inline marlowe-test/test/Spec/Marlowe/Marlowe.hs:162]: I've made a note to add this test into `marlowe-plutus`, if it is not already there.

**Review by bwbush — APPROVED** [review]
Looks good.

Also, I verified that the Marlowe validator scripts were transferred verbatim from https://github.com/input-output-hk/marlowe-cardano/commit/07ef19cdf5064363ae901f7768f39a128d14e407 to https://github.com/input-output-hk/marlowe-plutus/commit/4d0473e89676c4b97cd5d9a272f6cd06cca45822.

**Comment by bwbush** [comment]
> Question - what to do with the test report? The test code is now mostly in the other repo.

The test report was mostly for the auditors, and it hasn't been maintained since then. I'm wondering if we should delete it. What do you think, @palas?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -13,8 +13,27 @@ jobs:
...
+          remove-docker-images: 'true'
       - name: Checkout the code
         uses: actions/checkout@v3
+        with:
+          # Deep fetch to get the commit history to rewrite file mod times
+          fetch-depth: 0
+      - name: Rewrite file mod times (GHC uses file modification times for caching)
+        run: |
+          rev=HEAD
+          for f in $(git ls-tree -r -t --full-name --name-only "$rev") ; do
```
Comment (jhbertra) [inline .github/workflows/test.yml:34]:
```suggestion
          for f in "$(git ls-tree -r -t --full-name --name-only "$rev")" ; do
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -13,8 +13,27 @@ jobs:
...
         uses: actions/checkout@v3
+        with:
+          # Deep fetch to get the commit history to rewrite file mod times
+          fetch-depth: 0
+      - name: Rewrite file mod times (GHC uses file modification times for caching)
+        run: |
+          rev=HEAD
+          for f in "$(git ls-tree -r -t --full-name --name-only "$rev")" ; do
+              touch -d $(git log --pretty=format:%cI -1 "$rev" -- "$f") "$f";
+          done
```
Comment (jhbertra) [inline .github/workflows/test.yml:36]:
```suggestion
          IFS=$'\n'
          for f in $(git ls-tree -r -t --full-name --name-only "$rev") ; do
              touch -d $(git log --pretty=format:%cI -1 "$rev" -- "$f") "$f";
          done
```

**Pull request merged**